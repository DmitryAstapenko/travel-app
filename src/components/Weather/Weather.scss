$shadow-sm: 0 0.125rem 0.25rem rgba(0, 0, 0, 0.075);

.weather {
  position: relative;
  box-shadow: $shadow-sm;
  height: 140px;
  outline: none;
  cursor: default;
  opacity: 0.8;
  display: flex;
  width: 260px;
  background: #fff;
  border-radius: 5px;
  & .error {
    color: red;
  }
  & h4 {
    font-size: 15px;
    margin: 0;
  }
  & h1 {
    text-align: center;
    margin-bottom: 20px;
    margin-top: 10px;
  }
  &__left {
    width: 60%;
    padding: 10px;
  }
  &__right {
    width: 42%;
    display: flex;
    flex-direction: column;
    justify-content: space-evenly;
    align-items: center;
    background: #0000001f;
    text-align: center;
    padding: 10px;
    margin: 0;
  }
  & p {
    margin: 0;
    font-size: 13px;
  }
}

@media (max-width: 992px) {
  .weather.minimized {
    width: 50px;
    height: 50px;
    & .weather__left {
      display: none;
    }
    & .weather__right {
      background: none;
      transform: scale(0.6) translate(23px, 0);
      padding: 0;
    }

    cursor: pointer;
    height: 50px;
    h4 {
      display: none;
    }
  }
}

@media (max-width: 500px) {
  .weather {
    margin-top: 73px;
  }
}
