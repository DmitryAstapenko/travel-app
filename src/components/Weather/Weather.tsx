import axios from "axios";
import React, { createRef, FC, useContext, useEffect, useState } from "react";
// import ReactWeather, { useOpenWeather } from "react-open-weather";
import LocalizationContext from "../../context/LocalizationContext";
import ICountryData from "../../models/country-data";
import Loader from "../Loader";
import "./Weather.scss";

interface IWeatherProps {
  countryData: ICountryData | undefined;
}

interface IWeatherData {
  temp: number;
  id: string;
  humidity: number;
  wind: {
    deg: number;
    speed: number;
  };
}

const Weather: FC<IWeatherProps> = ({ countryData }: IWeatherProps) => {
  const language = useContext(LocalizationContext);
  const APIKey = "ce37a86730058fe8d88e2bd624a9e06f";
  const [weatherData, setWeatherData] = useState<IWeatherData | undefined>();
  // const weatherData = useOpenWeather({
  //   key: APIKey,
  //   lat: countryData?.capitalLocation.coordinates[0],
  //   lon: countryData?.capitalLocation.coordinates[1],
  //   lang: language,
  //   unit: "metric",
  // });
  // const { data, isLoading, errorMessage } = weatherData;
  // const customStyles = {
  //   fontFamily: "Helvetica, sans-serif",
  //   gradientStart: "white",
  //   gradientMid: "white",
  //   gradientEnd: "white",
  //   locationFontColor: "black",
  //   todayTempFontColor: "black",
  //   todayDateFontColor: "black",
  //   todayRangeFontColor: "black",
  //   todayDescFontColor: "black",
  //   todayInfoFontColor: "black",
  //   todayIconColor: "black",
  // };
  const [minimized, setMinimized] = useState("minimized");
  const clickHandler = (
    event: React.MouseEvent<HTMLDivElement, MouseEvent>
  ): void => {
    event.stopPropagation();
    setMinimized("");
  };
  const weatherRef = createRef<HTMLDivElement>();

  useEffect(() => {
    const closeWidgetes = () => {
      if (weatherRef.current) {
        setMinimized("minimized");
      }
    };
    window.addEventListener("click", closeWidgetes);
    return () => window.removeEventListener("click", closeWidgetes);
  });

  useEffect(() => {
    const getWeather = async () => {
      const res = await axios.get(
        `https://api.openweathermap.org/data/2.5/weather?q=${countryData?.localizations[language].capital}&lang=${language}&appid=${APIKey}&units=metric`
      );
      const {
        main: { temp, humidity },
        weather: [{ id }],
        wind,
      } = res.data;
      setWeatherData({
        temp,
        humidity,
        id,
        wind,
      });
    };

    getWeather();
  }, [countryData, language]);

  console.log(weatherData ?? "noWdata");

  return (
    // eslint-disable-next-line jsx-a11y/click-events-have-key-events
    <div
      ref={weatherRef}
      role="button"
      tabIndex={0}
      onClick={clickHandler}
      className={`weather ${minimized}`}
    >
      {/* <ReactWeather
        isLoading={isLoading}
        errorMessage={errorMessage}
        data={data}
        lang={language}
        locationLabel={countryData?.localizations.en.capital}
        unitsLabels={{ temperature: "â„ƒ", windSpeed: "Km/h" }}
        showForecast={false}
        theme={customStyles}
      /> */}
      {weatherData ? (
        <>
          <div className="weather__left">
            <h3>{weatherData.temp}</h3>
            <br />
            {/* <p>{weatherData.de}</p> */}
          </div>
          <div className="weather__icon">
            <i className={`owf owf-${weatherData.id}-d owf-5x`} />
          </div>
        </>
      ) : (
        <Loader />
      )}
    </div>
  );
};

export default Weather;
