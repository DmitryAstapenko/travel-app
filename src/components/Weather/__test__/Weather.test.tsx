import React from "react";
import Weather from "../Weather";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";

const setUp = (props) => shallow(<Weather {...props} />);

describe("should render weather component", () => {
  let component;
  beforeEach(() => {
    const countryData = {
      _id: "6048b7cc835218486b15e2c9",
      capitalLocation: {
        coordinates: [41.902782, 12.496365],
        type: "Point",
      },
      imageUrl:
        "https://www.atlanticcouncil.org/wp-content/uploads/2020/09/Rome-coroavirus-large-1024x683.jpg",
      videoUrl: "https://youtu.be/FlRwssZYRM0",
      currency: "EUR",
      ISOCode: "IT",
      flag: "https://restcountries.eu/data/ita.svg",
      timezones: ["UTC+01:00"],
      localizations: {
        en: {
          capital: "Rome",
          description:
            "Italy (Italian: Italia [iˈtaːlja]), officially the Italian Republic (Italian: Repubblica Italiana [reˈpubːlika itaˈljaːna]), is a country consisting of a peninsula delimited by the Alps and several islands surrounding it. Italy is located in Southern Europe, and is also considered part of Western Europe. A unitary parliamentary republic with Rome as its capital, the country covers a total area of 301,340 km2 (116,350 sq mi) and shares land borders with France, Switzerland, Austria, Slovenia, and the enclaved microstates of Vatican City and San Marino. Italy has a territorial enclave in Switzerland (Campione) and a maritime exclave in Tunisian waters (Lampedusa). With around 60 million inhabitants, Italy is the third-most populous member state of the European Union.",
          name: "Italy",
        },
        ru: {
          capital: "Рим",
          description:
            "Ита́лия (итал. Italia [iˈtaːlja]), официальное название — Италья́нская Респу́блика (итал. Repubblica Italiana [reˈpubːlika itaˈljaːna]) — государство в Южной Европе, в центре Средиземноморья. Входит в Евросоюз и НАТО с момента их создания, является третьей по величине экономикой еврозоны. Граничит с Францией на северо-западе (протяжённость границы — 488 км), Швейцарией (740 км) и Австрией (430 км) — на севере, Словенией — на северо-востоке (232 км). Внутри территории Италии находятся два государства-анклава: государство Сан-Марино и расположенное внутри территории Рима ассоциированное с Италией государство Ватикан, с каждым из которых Италия имеет внутреннюю границу протяжённостью соответственно 39 км и 3,2 км. Занимает Апеннинский полуостров, крайний северо-запад Балканского полуострова, Паданскую равнину, южные склоны Альп, острова Сицилия, Сардиния и ряд мелких островов. На территории Италии находится 55 памятников Всемирного наследия ЮНЕСКО — Италия разделяет с Китаем первое место по их количеству.",
          name: "Италия",
        },
        de: {
          capital: "Rom",
          description:
            "Italien, europäisches Land mit langer Mittelmeerküste, hat die westliche Kultur und Küche nachhaltig geprägt. In der Hauptstadt Rom befinden sich neben dem Vatikan auch imposante Kunstwerke und Ruinen antiker Bauwerke. Weitere bedeutende Städte des Landes sind Florenz, in dem Meisterwerke der Renaissance wie Michelangelos „David“ und Brunelleschis Duomo zu finden sind, Venedig, die Stadt der Kanäle, und Mailand, die Modehauptstadt Italiens.",
          name: "Italien",
        },
      },
    };
    component = setUp({ countryData });
  });

  it("should render weather component", () => {
    expect(toJson(component)).toMatchSnapshot();
  });

  it("should contain div class weather", () => {
    const weather = component.find(".weather");
    expect(weather).toHaveLength(1);
  });

  it("should contain div with class minimized", () => {
    const weather = component.find(".minimized");
    expect(weather.length).toEqual(1);
  });
});
