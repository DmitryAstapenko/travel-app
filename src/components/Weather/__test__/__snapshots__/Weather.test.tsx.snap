// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render weather component should render weather component 1`] = `
<div
  className="weather minimized"
  onClick={[Function]}
  role="button"
  tabIndex={0}
>
  <div
    className="mini-loader"
  >
    <Spinner
      animation="border"
    />
  </div>
</div>
`;
